subprojects/gdk_rust/gdk_electrum/src/interface.rs

```diff
@@ -41,27 +41,56 @@ pub struct WalletCtx {
     pub change_max_deriv: u32,
 }
 
-#[derive(Clone)]
+#[derive(Clone, Serialize, Deserialize, Debug)]
 pub enum ElectrumUrl {
     Tls(String, bool), // the bool value indicates if the domain name should be validated
     Plaintext(String),
 }
 
 impl ElectrumUrl {
     pub fn build_client(&self) -> Result<Client, Error> {
-        match self {
+        self.build_config(ConfigBuilder::new())
+    }
+
+    pub fn build_config(&self, config: ConfigBuilder) -> Result<Client, Error> {
+        let (url, config) = match self {
             ElectrumUrl::Tls(url, validate) => {
-                let client = RawClient::new_ssl(url.as_str(), *validate)?;
-                Ok(Client::SSL(client))
-            }
-            ElectrumUrl::Plaintext(url) => {
-                let client = RawClient::new(&url)?;
-                Ok(Client::TCP(client))
+                (format!("tls://{}", url), config.validate_domain(*validate))
```

Comment:
This must be `"ssl://{}"`

https://docs.rs/electrum-client/0.4.0-beta.1/electrum_client/client/struct.Client.html#method.new

Reply:
Good catch, thanks! Not the first time I get bitten by `tls` vs `ssl` 😅

Fixed in 6fdcd995d0df7b735721949996ec5cb9e184d2ee